src/hammer-vlsi/technology/asap7/__init__.py

```diff
@@ -212,9 +213,30 @@ def asap7_innovus_settings(ht: HammerTool) -> bool:
     ''')
     return True
 
-def scale_final_gds(ht: HammerTool) -> bool:
-    assert isinstance(ht, HammerPlaceAndRouteTool), "scale_final_gds can only run on par"
-    assert isinstance(ht, TCLTool), "scale_final_gds can only run on TCL tools"
+def asap7_update_floorplan(ht: HammerTool) -> bool:
+    assert isinstance(ht, HammerPlaceAndRouteTool), "asap7_update_floorplan can only run on par"
+    assert isinstance(ht, TCLTool), "asap7_update_floorplan can only run on TCL tools"
+    """
+    This is needed to move the core up by 1 site and re-do wiring tracks.
+    This resolves many DRCs and removes the need for the user to do it in placement constraints.
+    """
+    ht.append('''
+# Need to delete and recreate tracks based on tech LEF
+add_tracks -honor_pitch
+
+# Create place blockage on bottom row, fixes wiring issue + power vias for LVS
+set core_lly [get_db current_design .core_bbox.ll.y]
```

Comment:
This is interesting, and make me wonder what would happen if the user did this? Does it matter if the number of core sites is even or odd? Why does the issue not occur at the top?

Reply:
What do you mean if the user did this? Like, they created another blockage in the same place?

No, it doesn't matter if the number of rows is even or odd since it only places a blockage over the bottom-most row. For some reason that is unexplained, a variation of this is present in the reference script for ASAP7 and without it, everything shorts in the bottom row.